docs/architecture/0e-plans.md

```diff
@@ -568,6 +568,513 @@ Note that if we want to be able to remove all error code translations (to save c
+
+* Making the short-term workaround of enabling counter-direction functions official.
+* Having `MBEDTLS_X509_USE_C` require `MBEDTLS_ASN1_WRITE_C` in addition to `MBEDTLS_ASN1_PARSE_C`, and having `MBEDTLS_X509_CREATE_C` require `MBEDTLS_ASN1_PARSE_C` in addition to `MBEDTLS_ASN1_WRITE_C`.
+* Changing the X.509 code to reimplement the logic it needs, rather than use ASN.1 functions which don't fit the calling code well.
+* One of the above solutions for USE and a different one for CREATE.
+
+#### OID conditional compilation
+
+The compilation option `MBEDTLS_OID_C` guards the whole OID module (`oid.c`). Some OID table and support functions are further guarded according to the cryptographic mechanisms and library interfaces that are enabled. However, some OIDs that are generically useful in X.509 and not used directly by any other crypto code (e.g. `MBEDTLS_OID_X509_EXT_xxx`) are always included when `MBEDTLS_OID_C` is enabled. Furthermore, many OIDs for X.509 are guarded by `!defined(MBEDTLS_X509_REMOVE_INFO)`, which is a not a crypto option and therefore cannot appear in TF-PSA-Crypto (since we want to make it possible to build TF-PSA-Crypto independently of the crypto implementation).
+
+From the perspective of the TF-PSA-Crypto build, there are three levels of OID inclusion:
+
+1. Only as needed in crypto.
+2. (1) plus the core set for X.509 (subject to available cryptographic mechanisms where relevant).
+3. (2) plus the extra data for `mbedtls_x509_crt_info()` and friends.
+
+We either need public compilation options with this level of inclusion, or to move the X.509 parts of `oid.c` to the X.509 library (obviously preferable, but requiring more upfront work). TODO
+
+### Changes to option implications
+
+#### Changes to ASN.1 enablement
+
+Currently, `MBEDTLS_ASN1_PARSE_C` and `MBEDTLS_ASN1_WRITE_C` are automatically enabled:
+
+* If `MBEDTLS_RSA_C` is enabled. This is needed for RSA key import and export. We don't try to optimize code size when import or export of a key type is disabled, so this can stay. As part of internal refactoring, it will change to `PSA_WANT_KEY_TYPE_RSA_PUBLIC_KEY` eventually.
+* If `PSA_WANT_ALG_ECDSA` or `PSA_WANT_ALG_DETERMINISTIC_ECDSA` is built in. This used to be required to convert between ASN.1 and raw signature formats internally back when PSA was built on top of the legacy API, but today I think it isn't needed. However, it is needed for `MBEDTLS_ECDSA_C`. We do expose the signature conversion functions `mbedtls_ecdsa_raw_to_der` and `mbedtls_ecdsa_der_to_raw` when an ECDSA signature is enabled through PSA (built-in or not), but we automatically enable the necessary ASN.1 functions regardless of the ASN.1 configuration.
+
+So we have an optimization here to not auto-enable all of ASN.1 for built-in ECDSA. This will require removing or changing the guards on `mbedtls_ecdsa_read_signature()` and `mbedtls_ecdsa_write_signature()` and friends, which are now private functions.
```

Comment:
Just to be sure: to me this reads like no action required for $0\epsilon$ - just a possible improvement for 1.0 or even 1.x. Right?

Reply:
Yes, that would be a code size optimization and cleanup, which can be done in 1.x.